module/VuFind/src/VuFind/Config/Upgrade.php

```diff
@@ -1496,7 +1496,7 @@ protected function extractComments($filename)
                 // Is the current line a setting?  If so, add to the return value:
                 $set = trim(substr($trimmed, 0, $equals));
                 $set = trim(str_replace('[]', '', $set));
-                if (!empty($section) && !empty($set)) {
+                if ('' == $section && '' == $set) {
```

Comment:
This seems to be reversing the logic. If this didn't cause a test to fail, maybe we'd better add another test.

Reply:
Oops, fixed! It'd only affect inline comments, but yes, not caught by any test.

Reply:
I've just pushed up a test to cover this code, and confirmed that it failed before your fix and passes after.